cmd/snap-confine/mount-support.c

```diff
@@ -217,6 +217,7 @@ struct sc_mount_config {
 	// The struct is terminated with an entry with NULL path.
 	const struct sc_mount *mounts;
 	bool on_classic_distro;
+	bool uses_base_snap;
```

Comment:
I'd like to have code that makes this work without any flags. In essence all snaps use base snap so our approach should be more universal.

Reply:
The flag is currently only used to ensure `/usr/lib/snapd` (and later `snapctl`) is available for the snap. We need /usr/lib/snapd/snap-exec to actually running anything in a base so that is important. On a normal core system we do not have to make /usr/lib/snapd available, its already there.